pkg/agent/entrypoint.go

```diff
@@ -72,7 +70,7 @@ func (r MainRunner) Run() error {
 			globalState.Stop()
 			return nil
 		case event := <-podEvents:
-			globalState.handleEvent(event)
+			globalState.handleEvent(ctx, event)
```

Comment:
this feels icky (`handleEvent` will spawn threads using the context long after `handleEvent` finishes, and we're always using the same context for `handleEvent`) - but the only better solution is storing the context in `globalstate` itself.

Thoughts?

Reply:
I don't think there's any problem with passing a context to a function that returns before the goroutines it spawns (and indeed having this context means that shutting down the agents main loop will actually cause a shutdown. Eventually/soon the basecontext/shutdown stuff can/will help make some of this more manageable.